### ISSUE TITLE
prov/opx: compilation failed on ARM instances (because of fi_opx_timer_get_cycles)

### ISSUE THREAD

**Issue by wzamazon** [ofiwg/libfabric#7573]
**Describe the bug**

libfabric  compilation currently will fail on ARM instance if opx provider is enabled.

**To Reproduce**

```
./configure
make
```
Error message:
```
[](https://hpc-ci.aws.a2z.com/blue/organizations/jenkins/NightlyC6gnTests/detail/NightlyC6gnTests/90/pipeline#step-53-log-2179)[](https://hpc-ci.aws.a2z.com/blue/organizations/jenkins/NightlyC6gnTests/detail/NightlyC6gnTests/90/pipeline#step-53-log-2180)[](https://hpc-ci.aws.a2z.com/blue/organizations/jenkins/NightlyC6gnTests/detail/NightlyC6gnTests/90/pipeline#step-53-log-2181)[](https://hpc-ci.aws.a2z.com/blue/organizations/jenkins/NightlyC6gnTests/detail/NightlyC6gnTests/90/pipeline#step-53-log-2182)[](https://hpc-ci.aws.a2z.com/blue/organizations/jenkins/NightlyC6gnTests/detail/NightlyC6gnTests/90/pipeline#step-53-log-2183)[](https://hpc-ci.aws.a2z.com/blue/organizations/jenkins/NightlyC6gnTests/detail/NightlyC6gnTests/90/pipeline#step-53-log-2184)[](https://hpc-ci.aws.a2z.com/blue/organizations/jenkins/NightlyC6gnTests/detail/NightlyC6gnTests/90/pipeline#step-53-log-2185)[](https://hpc-ci.aws.a2z.com/blue/organizations/jenkins/NightlyC6gnTests/detail/NightlyC6gnTests/90/pipeline#step-53-log-2186)[](https://hpc-ci.aws.a2z.com/blue/organizations/jenkins/NightlyC6gnTests/detail/NightlyC6gnTests/90/pipeline#step-53-log-2187)[](https://hpc-ci.aws.a2z.com/blue/organizations/jenkins/NightlyC6gnTests/detail/NightlyC6gnTests/90/pipeline#step-53-log-2188)[](https://hpc-ci.aws.a2z.com/blue/organizations/jenkins/NightlyC6gnTests/detail/NightlyC6gnTests/90/pipeline#step-53-log-2189)[](https://hpc-ci.aws.a2z.com/blue/organizations/jenkins/NightlyC6gnTests/detail/NightlyC6gnTests/90/pipeline#step-53-log-2190)[](https://hpc-ci.aws.a2z.com/blue/organizations/jenkins/NightlyC6gnTests/detail/NightlyC6gnTests/90/pipeline#step-53-log-2191)[](https://hpc-ci.aws.a2z.com/blue/organizations/jenkins/NightlyC6gnTests/detail/NightlyC6gnTests/90/pipeline#step-53-log-2192)[](https://hpc-ci.aws.a2z.com/blue/organizations/jenkins/NightlyC6gnTests/detail/NightlyC6gnTests/90/pipeline#step-53-log-2193)[](https://hpc-ci.aws.a2z.com/blue/organizations/jenkins/NightlyC6gnTests/detail/NightlyC6gnTests/90/pipeline#step-53-log-2194)make[1]: *** [prov/opx/src/src_libfabric_la-fi_opx_av.lo] Error 1

make[1]: *** Waiting for unfinished jobs....

make[1]: *** [prov/opx/src/src_libfabric_la-fi_opx_atomic.lo] Error 1

In file included from ./prov/opx/include/rdma/opx/fi_opx_reliability.h:44:0,

                 from ./prov/opx/include/rdma/opx/fi_opx_domain.h:44,

                 from prov/opx/src/fi_opx_cm.c:35:

./prov/opx/include/rdma/opx/fi_opx_timer.h:64:2: error: #error "Cycle timer not defined for this platform"

 #error "Cycle timer not defined for this platform"

  ^~~~~

./prov/opx/include/rdma/opx/fi_opx_timer.h: In function 'fi_opx_timer_init':

./prov/opx/include/rdma/opx/fi_opx_timer.h:86:20: warning: implicit declaration of function 'fi_opx_timer_get_cycles'; did you mean 'fi_opx_timer_init'? [-Wimplicit-function-declaration]

  uint64_t cycles = fi_opx_timer_get_cycles();

                    ^~~~~~~~~~~~~~~~~~~~~~~

                    fi_opx_timer_init

make[1]: *** [prov/opx/src/src_libfabric_la-fi_opx_cm.lo] Error 1

make[1]: Leaving directory `/fsx/libfabric'

make: *** [all] Error 2
```

**Expected behavior**
opx provider disable itself on ARM platform.

**Output**
If applicable, add output to help explain your problem. (e.g. backtrace, debug logs)

**Environment:**
Amazon linux2

**Comment by j-xiong** [comment]
@mwheinz Does omni-path support any architecture other than x86_64?

**Comment by mwheinz** [comment]
> @mwheinz Does omni-path support any architecture other than x86_64?

@j-xiong - no. I thought it was already disabled from building on anything except x86_64.

**Comment by wzamazon** [comment]
linux.

specifically, amazon linux 2 on AWS c6gn.16xlarge instance (which is an ARM based instance).

**Comment by mwheinz** [comment]
> specifically, amazon linux 2 on AWS c6gn.16xlarge instance (which is an ARM based instance).

Fair enough. Stupid of me not to realize. I assumed that mimicking how the PSM2 provider gets configured would be sufficient, turns out that PSM2 does actually compile on ARM even though you'll never find OPA hardware on an ARM machine!

I should have a PR for this and the other build issue in an hour or two - testing it now.

**Comment by j-xiong** [comment]
Fixed by #7583. Not sure why it is not closed automatically with the "Fixes" phrase.
